contracts/ReputationMiningCycleRespond.sol

```diff
@@ -0,0 +1,387 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity ^0.4.23;
+pragma experimental "v0.5.0";
+
+import "../lib/dappsys/math.sol";
+import "./IColonyNetwork.sol";
+import "./PatriciaTree/PatriciaTreeProofs.sol";
+import "./ITokenLocking.sol";
+import "./ReputationMiningCycleStorage.sol";
+
+
+// TODO: Can we handle all possible disputes regarding the very first hash that should be set?
+// Currently, at the very least, we can't handle a dispute if the very first entry is disputed.
+// A possible workaround would be to 'kick off' reputation mining with a known dummy state...
+contract ReputationMiningCycleRespond is ReputationMiningCycleStorage, PatriciaTreeProofs, DSMath {
+
+  /// @notice A modifier that checks if the challenge corresponding to the hash in the passed `round` and `id` is open
+  /// @param round The round number of the hash under consideration
+  /// @param idx The index in the round of the hash under consideration
+  modifier challengeOpen(uint256 round, uint256 idx) {
+    // TODO: More checks that this is an appropriate time to respondToChallenge
+    require(disputeRounds[round][idx].lowerBound == disputeRounds[round][idx].upperBound, "colony-reputation-mining-challenge-closed");
+    _;
+  }
+
+  uint constant U_ROUND = 0;
+  uint constant U_IDX = 1;
+  uint constant U_REPUTATION_BRANCH_MASK = 2;
+  uint constant U_AGREE_STATE_NNODES = 3;
+  uint constant U_AGREE_STATE_BRANCH_MASK = 4;
+  uint constant U_DISAGREE_STATE_NNODES = 5;
+  uint constant U_DISAGREE_STATE_BRANCH_MASK = 6;
+  uint constant U_PREVIOUS_NEW_REPUTATION_BRANCH_MASK = 7;
+  uint constant U_REQUIRE_REPUTATION_CHECK = 8;
+  uint constant U_LOG_ENTRY_NUMBER = 9;
+  uint constant U_DECAY_TRANSITION = 10;
+
+  function respondToChallenge(
```

Comment:
`solium` complains about a blank line rule broken here

Reply:
Fixed, but this was only a warning, which is why linting passed on the precommit hook. Should we make it an error?

Reply:
I have been going for selectively switching _on_ `solium` warning rules to errors, like line length for example but I find we need to go for maximum strictness on contracts linting so I like the suggestion of just making all warnings errors and selectively switch _off_ rules if we feel the need.